src/taco-remote-lib/ios/ios.ts

```diff
@@ -113,37 +118,61 @@ class IOSAgent implements ITargetPlatform {
             callback(msg);
             return;
         }
-
-        var archive: Packer = new Packer();
-        archive.on("error", function (err: Error): void {
-            Logger.logError(resources.getString("ArchivePackError", err.message));
-            callback(err);
-            res.status(404).send(resources.getStringForLanguage(req, "ArchivePackError", err.message));
-        });
-        res.set({ "Content-Type": "application/zip" });
-        archive.pipe(res);
-        archive.entry(fs.createReadStream(pathToPlistFile), { name: buildInfo["appName"] + ".plist" },
-            function (pListEntryError: Error, pListEntry: any): void {
-            if (pListEntryError) {
-                Logger.logError(resources.getString("ArchivePackError", pListEntryError.message));
-                callback(pListEntryError);
-                res.status(404).send(resources.getStringForLanguage(req, "ArchivePackError", pListEntryError.message));
-                return;
-            }
-
-            archive.entry(fs.createReadStream(pathToIpaFile), { name: buildInfo["appName"] + ".ipa" },
-                function (ipaEntryError: Error, ipaEntry: any): void {
-                if (ipaEntryError) {
-                    Logger.logError(resources.getString("ArchivePackError", ipaEntryError.message));
-                    callback(ipaEntryError);
-                    res.status(404).send(resources.getStringForLanguage(req, "ArchivePackError", ipaEntryError.message));
-                    return;
-                }
-
+        
+        Q({})
+            .then(function(): Q.Promise<any> {
+                var deferred: Q.Deferred<any> = Q.defer();
+                
+                var outputStream = fs.createWriteStream(pathToDsymZipFile);
+                var archive: any = archiver("zip");
+                archive.on("error", function (err: Error): void {
+                    return deferred.reject(err);
+                });
+                archive.on("finish", function() {
+                    return deferred.resolve({});
```

Comment:
I'm not sure, but there may be a slight race condition here: You are resolving the deferred when the archive stream finishes writing to the outputStream, but the outputStream may not yet have finished writing to disk. Might be best to resolve on outputStream's `end` event to guarantee that the next promise step can find the completed file on disk?